### ISSUE TITLE
aws s3 sync should retry multipart uploads that fail on ECONNRESET

### ISSUE THREAD

**Issue by alex** [aws/aws-cli#4554]
If you're uploading a sufficiently large file, on a sufficiently spotty network, you're bound to get an ECONNRESET sooner or later.

It'd be great if `aws s3 sync` was resllient to this and could retry multipart uploads which error'd.

This is basically the same issue as https://github.com/aws/aws-sdk-go/issues/2737

**Comment by joguSD** [comment]
I agree this would be a good thing to retry. Marking this as a feature request.
